### PR TITLE
Include node namespaces in get_node_names()

### PR DESCRIPTION
**Pull request by mjcarroll**
Connects to ros2/rmw#142

### CODE REVIEW

File: rmw/include/rmw/rmw.h [hunk 1]
```diff
@@ -473,7 +473,8 @@ RMW_WARN_UNUSED
 rmw_ret_t
 rmw_get_node_names(
   const rmw_node_t * node,
-  rcutils_string_array_t * node_names);
+  rcutils_string_array_t * node_names,
+  rcutils_string_array_t * node_namespaces);
```
Comment (dirk-thomas) [inline rmw/include/rmw/rmw.h]: With the signature becoming more complicated can you please add a docblock to document the semantic of these arguments.

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5115/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1895/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4241/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5092/)

File: rmw/include/rmw/rmw.h [hunk 2]
```diff
@@ -468,12 +468,37 @@ rmw_wait(
   rmw_wait_set_t * wait_set,
   const rmw_time_t * wait_timeout);
 
+/// Return a list of node name and namespaces discovered via a node.
+/**
+ * This function will return a list of node names and a list of node namespaces
+ * that are discovered via the middleware. The list of names and namespaces
+ * will be the same length.
```
Comment (dirk-thomas) [inline rmw/include/rmw/rmw.h]: While kind of obvious maybe the doc should be explicit that these two lists represent pairs of namespace and name for each node.

File: rmw/include/rmw/rmw.h [hunk 3]
```diff
@@ -468,12 +468,40 @@ rmw_wait(
   rmw_wait_set_t * wait_set,
   const rmw_time_t * wait_timeout);
 
+/// Return a list of node name and namespaces discovered via a node.
```
Comment (wjwwood) [inline rmw/include/rmw/rmw.h]: nitpick: either `node names and namespaces` or `node name and namespace pairs`?

**Comment by mjcarroll** [comment]
Enough changes to justify running CI again:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5137/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1915/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4262/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5116/)

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5137/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1915/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4262/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5116/)

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5138/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1916/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4263/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5117/)

**Comment by mjcarroll** [comment]
I fixed the new warnings in `rmw_fastrtps`

**Pull request merged**